include/aws/mqtt/private/request-response/protocol_adapter.h

```diff
@@ -0,0 +1,196 @@
+#ifndef AWS_MQTT_PRIVATE_REQUEST_RESPONSE_PROTOCOL_ADAPTER_H
+#define AWS_MQTT_PRIVATE_REQUEST_RESPONSE_PROTOCOL_ADAPTER_H
+
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/mqtt/exports.h>
+#include <aws/mqtt/mqtt.h>
+
+#include <aws/common/byte_buf.h>
+
+struct aws_allocator;
+struct aws_mqtt_client_connection;
+struct aws_mqtt5_client;
+
+/*
+ * The request-response protocol adapter is a translation layer that sits between the request-response native client
+ * implementation and a protocol client capable of subscribing, unsubscribing, and publishing MQTT messages.
+ * Valid protocol clients include the CRT MQTT5 client, the CRT MQTT311 client, and an eventstream RPC connection
```

Comment:
Trivial: Should we exclude eventstream RPC from the description until it's implemented?

Reply:
I don't see the need.  This is in a private and will get moved to a different repository eventually anyways.